lib/kafka/producer.rb

```diff
@@ -257,6 +257,25 @@ def buffer_bytesize
       @pending_message_queue.bytesize + @buffer.bytesize
     end
 
+    # The messages waiting to get delivered.
+    #
+    # @return [Array<String>]
+    def buffer_messages
+      messages = []
+
+      @buffer.each do |topic, partition, messages_for_partition|
+        messages_for_partition.each do |message|
+          messages << message.value
+        end
+      end
+
+      @pending_message_queue.each do |message|
+        messages << message.value
```

Comment:
It's not enough to return the `value`. This PR would necessitate making some sort of message object part of the public API, which I'm not too happy about.

Reply:
:+1: That makes sense